src/monodroid/jni/monodroid-glue.c

```diff
@@ -704,55 +737,192 @@ typedef void* (*mono_mkbundle_init_ptr) (void (*)(const MonoBundledAssembly **),
 mono_mkbundle_init_ptr mono_mkbundle_init;
 void (*mono_mkbundle_initialize_mono_api) (const BundleMonoAPI *info);
 
-static void
-setup_bundled_app (const char *libappso)
+static char*
+get_full_dso_path (const char *base_dir, const char *dso_path, mono_bool *needs_free)
```

Comment:
Should we try to move these new DSO functions to a new file? `monodroid-glue.c` is getting rather large; can we avoid making it worse?  (*Should* we avoid making it "worse"?  Would keeping it in this file be better?)

Reply:
I think we should move them to a new file, but I'd like to merge https://github.com/xamarin/xamarin-android/pull/2089 first